secret-manager/runtime/src/main/java/io/quarkiverse/googlecloudservices/secretmanager/runtime/config/SecretManagerConfigSource.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkiverse.googlecloudservices.secretmanager.runtime.config;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.google.cloud.secretmanager.v1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+
+import io.quarkus.arc.Arc;
+import io.smallrye.config.common.AbstractConfigSource;
+
+public class SecretManagerConfigSource extends AbstractConfigSource {
+
+    private static final String CONFIG_SOURCE_NAME = "io.quarkiverse.googlecloudservices.secretmanager.runtime.config";
+
+    private final String defaultProject;
+
+    // This will be initialized the first time `getValue` is called.
+    private SecretManagerServiceClient client = null;
+
+    public SecretManagerConfigSource(String defaultProject) {
+        super(CONFIG_SOURCE_NAME, DEFAULT_ORDINAL);
```

Comment:
Maybe it needs to use a lower ordinal to avoid being called with propertyNames from application.properties

Reply:
Done. -- Set to 50. Seems like any value less than 100 is okay; I just chose 50 since it gives some room for future contributors to put things before or after.

Reply:
Thanks, I'll ask a review from someone working on Smallrye Config to validate this.